geode-core/src/main/java/org/apache/geode/internal/cache/versions/RegionVersionVector.java

```diff
@@ -994,13 +994,18 @@ public boolean isNewerThanOrCanFillExceptionsFor(RegionVersionVector<T> other) {
     return false;
   }
 
-  private boolean isGCVersionDominatedByOtherHolder(Long gcVersion,
+  private boolean isGCVersionDominatedByOtherHolder(Long gcVersion, Long requesterGCVersion,
       RegionVersionHolder<T> otherHolder) {
     if (gcVersion == null || gcVersion.longValue() == 0) {
       return true;
     } else {
       RegionVersionHolder<T> holder = new RegionVersionHolder<T>(gcVersion.longValue());
-      return !holder.isNewerThanOrCanFillExceptionsFor(otherHolder);
+      if (otherHolder == null) {
+        return false;
+      } else {
+        return otherHolder.dominates(holder)
+            && (gcVersion == requesterGCVersion || otherHolder.getVersion() > gcVersion);
```

Comment:
It seems like you are comparing gcversion to rvv in the  final comparison on this line 

> otherHolder.getVersion() > gcVersion

Reply:
that's correct